contents/tutorials/nextjs-ab-tests.md

```diff
@@ -0,0 +1,269 @@
+  })
+}
+
+export default function App({ Component, pageProps }) {
+  return (
+    <PostHogProvider client={posthog}>
+      <Component {...pageProps} />
+    </PostHogProvider>
+  )
+}
+```
+
+Once you’ve done this, reload your app and click the button a few times. You should see captured in PostHog.
+
+![Events](../images/tutorials/nextjs-ab-tests/events.png)
+
+## Creating an action for our experiment goal
+
+The first part of setting up our experiment in PostHog is setting up the goal. In this case, it is clicks of the main call to action button which has varying content.
+
+To measure this, we can set up an [action](/manual/actions). We can use the PostHog toolbar to create this. To enable and launch the toolbar, go to the "Launch Toolbar" tab in PostHog, add `http://localhost:3000/` as an authorized URL, then click launch. This brings you to your app with a PostHog icon hovering on your screen.
+
+To create an action with the toolbar, click:
+
+1. the PostHog icon
+2. "Inspect" 
+3. the "Click Me" button, 
+4. "Create New Action" in the modal. 
+
+Name the action "Clicked Main CTA" and unselect the text match (because it changes), then scroll to the bottom of the modal to create an action.
+
+![Action](../images/tutorials/nextjs-ab-tests/action.gif)
+
+## Creating an experiment
+
+With our Next.js app, PostHog, and our experiment goal set up, we can go into PostHog, and create a new experiment.
+
+> Experiments are a paid feature, but if you are under 1M events and 15k session recordings per month, they are free.
+
+In your PostHog instance, go to the experiments tab, click create a new experiment, and enter a name and key. We keep the defaults for variants and not use any filters (which means 100% of users see the experiment). Use the action we created earlier as our experiment goal. 
+
+![Experiment](../images/tutorials/nextjs-ab-tests/experiment.gif)
+
+> **Note:** There must be at least one occurrence of the action for it to show as a goal, so make sure to click the button after creating the action.
+
+Once we create the experiment, we can start to implement it. Come back and launch it later.
+
+## Implementing our experiment
+
+When it comes to implementing our experiment and its associated feature flags, there are a few options we can choose:
+
+1. Bootstrap feature flags by using [`getInitialProps`](https://nextjs.org/docs/api-reference/data-fetching/get-initial-props) in `_app.js`
+2. Server-side render feature flags in `index.js` using `posthog-node`.
+3. Client-side render feature flags in `index.js`.
+
+Because `getInitialProps` disables [Automatic Static Optimization](https://nextjs.org/docs/advanced-features/automatic-static-optimization) for the entire app (one of the main benefits of Next.js), we aren’t going to bootstrap feature flags (option 1). Instead, we use a combination of server-side and client-side rendering. This ensures flags' accuracy and speed while still leveraging the unique benefits of Next.js.
```

Comment:
```suggestion
Because `getInitialProps` disables [Automatic Static Optimization](https://nextjs.org/docs/advanced-features/automatic-static-optimization) for the entire app (one of the main benefits of Next.js), we aren’t going to bootstrap feature flags (option 1). Instead, we use a combination of server-side and client-side rendering. This ensures the flags' accuracy and speed while still leveraging the unique benefits of Next.js.
```